src/Core/Components/DateTime/CalendarExtended.cs

```diff
@@ -159,36 +146,29 @@ public string GetYearsRangeLabel(int fromYear)
         var min = fromYear;
         var max = fromYear + 11;
 
-        if (min < DateTime.MinValue.Year)
+        var minSupportedYear = Culture.Calendar.MinSupportedDateTime.GetYear(Culture);
+        var maxSupportedYear = Culture.Calendar.MaxSupportedDateTime.GetYear(Culture);
+
+        if (min < minSupportedYear)
         {
-            min = DateTime.MinValue.Year;
+            min = minSupportedYear;
         }
 
-        if (max > DateTime.MaxValue.Year)
+        if (max > maxSupportedYear)
         {
-            max = DateTime.MaxValue.Year;
+            max = maxSupportedYear;
         }
 
         return min == max ? $"{min}" : $"{min} - {max}";
     }
 
-    /// <summary>
-    /// Returns the year value.
-    /// </summary>
-    /// <param name="date"></param>
-    /// <returns></returns>
-    public string GetYear(DateTime date)
-    {
-        return $"{date.Year}";
-    }
-
     /// <summary>
     /// Returns a list of days, abbreviated and complete (Mon, Monday), ...,(Sun, Sunday) in the correct culture.
     /// </summary>
     /// <returns></returns>
     public IEnumerable<(string Abbreviated, string Shorted, string Name)> GetDayNames()
     {
-        var firstDayOfWeek = (int)GetFirstDayOfWeek();
+        var firstDayOfWeek = Culture.DateTimeFormat.FirstDayOfWeek;
```

Comment:
Could you replace this line by `(int)Culture.DateTimeFormat.FirstDayOfWeek` to keep the same "logic" and to remove the `(int)` from `return Shift(dayNames, (int)firstDayOfWeek);`